### DISCUSSION TITLE
Enable versioning for online documentation using mike + mkdocs

### DISCUSSION THREAD

**Discussion by ghandic** [fastapi/fastapi#8657] [Questions]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
NA
```

### Description

Given there could be some breaking changes the would eventually happen (or already have) and people are relying on FastAPI for production services we should keep a record of the documentation for multiple versions of FastAPI.

If we think of the most basic problem, someone on an old version might took at the docs and think they can implement something that they cannot on the version they are pinned to. Likewise a feature could be depreciated, or people linking to the docs without a permalink from stack overflow to solve an issue that has now been removed from the docs.

### Wanted Solution

Utilize mike and have some agreed rules on which docs can be updated per version (eg only versioning the minor tag and not patch tags.) https://squidfunk.github.io/mkdocs-material/setup/setting-up-versioning/ 

### Wanted Code

```python
NA
```

### Operating System

Other

### Operating System Details

NA

### FastAPI Version

NA

### Python Version

NA

### Additional Context

Open to disucssion

**Comment by Kludex** [comment]
Maybe just versioning the API reference once available?
